core/src/Streamly/Data/Stream.hs

```diff
@@ -234,6 +234,7 @@ module Streamly.Data.Stream
     -- express all others.
     , fold -- XXX rename to run? We can have a Stream.run and Fold.run.
     -- XXX fold1 can be achieved using Monoids or Refolds.
+    -- XXX We can call this just "break" and parseBreak as "munch"
```

Comment:
What? `foldBreak` and `parseBreak` sound great. The names should be similar. Folds and parsers are sisters.